binaries/daemon/src/lib.rs

```diff
@@ -379,6 +388,54 @@ impl Daemon {
                 Event::DaemonError(err) => {
                     tracing::error!("Daemon error: {err:?}");
                 }
+                Event::SpawnNodeResult {
+                    dataflow_id,
+                    node_id,
+                    result,
+                } => match result {
+                    Ok(running_node) => {
+                        if let Some(dataflow) = self.running.get_mut(&dataflow_id) {
+                            dataflow.running_nodes.insert(node_id, running_node);
+                        } else {
+                            tracing::error!("failed to handle SpawnNodeResult: no running dataflow with ID {dataflow_id}");
+                        }
+                    }
+                    Err(error) => {
+                        self.dataflow_node_results
+                            .entry(dataflow_id)
+                            .or_default()
+                            .insert(node_id.clone(), Err(error));
+                        self.handle_node_stop(dataflow_id, &node_id).await?;
+                    }
+                },
+                Event::SpawnDataflowResult {
+                    dataflow_id,
+                    result,
+                } => {
+                    if let Some(connection) = &mut self.coordinator_connection {
+                        let msg = serde_json::to_vec(&Timestamped {
+                            inner: CoordinatorRequest::Event {
+                                daemon_id: self.daemon_id.clone(),
+                                event: DaemonEvent::SpawnResult {
+                                    dataflow_id,
+                                    result: result.map_err(|err| format!("{err:?}")),
+                                },
+                            },
+                            timestamp: self.clock.new_timestamp(),
+                        })?;
+                        socket_stream_send(connection, &msg)
+                            .await
+                            .wrap_err("failed to send Exit message to dora-coordinator")?;
+                    }
+                }
+            }
+
+            let elapsed = start.elapsed();
+            if elapsed > Duration::from_millis(100) {
+                tracing::warn!(
+                    "Daemon took {}ms for handling event: {event_debug}",
+                    elapsed.as_millis()
+                );
```

Comment:
This also look like debugging code?

Reply:
See my comment [above](https://github.com/dora-rs/dora/pull/901#discussion_r2056304232).